receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -91,6 +117,49 @@ func newAlertsReceiver(logger *zap.Logger, cfg AlertConfig, consumer consumer.Lo
 }
 
 func (a alertsReceiver) Start(ctx context.Context, host component.Host) error {
+	switch a.mode {
+	case alertModeListen:
+		return a.startListening(ctx, host)
+	case alertModeRetrieval:
+		return a.startRetrieving(ctx, host)
+	default:
+		return fmt.Errorf("receiver attempted to start without a known mode: %s", a.mode)
+	}
+}
+
+func (a alertsReceiver) startRetrieving(ctx context.Context, host component.Host) error {
+	client, err := internal.NewMongoDBAtlasClient(a.publicKey, a.privateKey, a.retrySettings, a.logger)
+	if err != nil {
+		return err
+	}
+	a.client = client
+
+	// may want to consider using disk storage for tracking which alerts have been sent
+	// or may want to revisit how we get alerts
+	// this is a memory cache used to keep track of unupdated alerts
+	cache, err := buntdb.Open(":memory:")
```

Comment:
I'm wondering if it is necessary to use this database library here; Seems kinda heavy-weight for the use-case (the TTL stuff is nice, but still seems overkill).


Another note; Later down the line, if we were to implement persistence, it'd be with https://github.com/open-telemetry/opentelemetry-collector/tree/main/extension/experimental/storage (filestorage, which implements this interface, is available in the contrib repo).